File: Python/statistics.py
import os


flips = [] # list of all flips
folder = os.listdir("C:\\Users\\PC\\Desktop\\Python\\coin flip\\Data") # folder with data
print(folder)

# opening every txt file in data folder and adding flips to the list

iteration = 1
for file in folder:
    with open (f"C:\\Users\\PC\\Desktop\\Python\\coin flip\\Data\\{file}", "r") as x:
        for line in x.readlines():
            flips.append(line)
# reformating the data to make everything look cleaner
data = []
for outcome in flips:
    WhatWeNeed = outcome.replace("\n","")
    data.append(WhatWeNeed)
heads = 0
hstreak = 0
hhstreak = 0
tails = 0
tstreak = 0
htstreak = 0

# counting throws and highest streak of both heads and tails

for record in data:
    if record == "Heads":
        heads += 1
        tstreak = 0
        hstreak += 1
        if hstreak > hhstreak:
            hhstreak = hstreak
    else:
        tails += 1
        hstreak = 0
        tstreak += 1
        if tstreak > htstreak:
            htstreak = tstreak

print("Number of times coin was flipped: " + str(len(data)))
print("Number of Heads: " + str(heads) + ", Number of Tails: " + str(tails))
print("Percent of Heads: " + str(heads/len(data)*100) + "%, Percent of Tails: " + str(tails/len(data)*100) + "%")
print("Highest consequtives Heads throws in a row: " + str(hhstreak) + ", Highest consequtives Heads throws in a row: " + str(htstreak))

File: C/statistics.c
#include <stdio.h>


int main(){
    FILE *data = fopen("data.txt", "r");

    if(data == NULL){
        fprintf(stderr, "Unable to open the file.\n");
        return 1;
    }

    int heads = 0, tails = 0;
    int hstreak = 0, tstreak = 0; // Current Streak
    int hhstreak = 0, htstreak = 0; // Highest Streak


    int flip, size = 0;

    while((flip = fgetc(data)) != EOF){
        if(flip == '\n'){
        continue;
        }
        if(flip == 49){
            tails++;
            hstreak = 0;
            tstreak++;
            if(tstreak > htstreak){
                htstreak = tstreak;
            }
        } else{
            heads++;
            tstreak = 0;
            hstreak++;
            if(hstreak > hhstreak){
                hhstreak = hstreak;
            }
        }
        size++;
    }

    fclose(data);

    float hpercent = (float)heads / size * 100, tpercent = (float)tails / size * 100;

    printf("Number of times coin was flipped: %d\n", size);
    printf("Number of Heads: %d, Number of Tails: %d\n", heads, tails);
    printf("Percent if Heads: %.2f%, Percent of Tails %.2f%\n", hpercent, tpercent);
    printf("Highest consequtive Heads throws in a row: %d, "
    "Highest consequtive Tails throws in a row: %d\n", hhstreak, htstreak);

    return 0;
}
